scripts/posteffects/posteffect-ssao.js

```diff
@@ -354,42 +352,114 @@ function SSAOEffect(graphicsDevice, ssaoScript) {
             "    // this is most useful with high quality/large blurs",
             "    // ao += ((random(gl_FragCoord.xy) - 0.5) / 255.0);",
             "",
-            "    vec4 inCol = texture2D( uColorBuffer,  vUv0 );",
-            "",
             "    ao = mix(ao, 1.0, uBrightness);",
-            "    gl_FragColor.rgb = inCol.rgb * ao;",
-            "    gl_FragColor.a = 1.0;",
+            "    gl_FragColor.a = ao;",
             "}"
         ].join("\n")
     });
 
+    this.outputShader = new pc.Shader(graphicsDevice, {
+        attributes: {
+            aPosition: pc.SEMANTIC_POSITION
+        },
+        vshader: [
+            (graphicsDevice.webgl2) ? ("#version 300 es\n\n" + pc.shaderChunks.gles3VS) : "",
+            graphicsDevice.webgl2 ? "" : "#extension GL_OES_standard_derivatives: enable\n",
+            "attribute vec2 aPosition;",
+            "",
+            "varying vec2 vUv0;",
+            "",
+            "void main(void)",
+            "{",
+            "    gl_Position = vec4(aPosition, 0.0, 1.0);",
+            "    vUv0 = (aPosition.xy + 1.0) * 0.5;",
+            "}"
```

Comment:
it seems this vertex shader is repeated 2-3x in this file .. perhaps store it in a variable and reuse?

Reply:
I am actually working on adding a createVertexShader for post effects so we can reuse it everywhere. It seems this code is the same for every post effect, so it would make sense to have it declared in a single place and reused across multiple post effects. I will add it to this PR.